subsys/net/lib/mqtt/mqtt_transport_socket_tcp.c

```diff
@@ -29,26 +32,44 @@ int mqtt_client_tcp_connect(struct mqtt_client *client)
 	const struct sockaddr *broker = client->broker;
 	int ret;
 
-	client->transport.tcp.sock = socket(broker->sa_family, SOCK_STREAM,
-					    IPPROTO_TCP);
-	if (client->transport.tcp.sock < 0) {
-		return -errno;
-	}
+#if defined(CONFIG_SOCKS) && !defined(CONFIG_MQTT_LIB_TLS)
+	if (client->proxy == NULL)
+#endif
+	{
+		client->transport.tcp.sock = socket(broker->sa_family,
+						    SOCK_STREAM,
+						    IPPROTO_TCP);
+		if (client->transport.tcp.sock < 0) {
+			return -errno;
+		}
 
-	MQTT_TRC("Created socket %d", client->transport.tcp.sock);
+		MQTT_TRC("Created socket %d", client->transport.tcp.sock);
 
-	size_t peer_addr_size = sizeof(struct sockaddr_in6);
+		size_t peer_addr_size = sizeof(struct sockaddr_in6);
 
-	if (broker->sa_family == AF_INET) {
-		peer_addr_size = sizeof(struct sockaddr_in);
+		if (broker->sa_family == AF_INET) {
+			peer_addr_size = sizeof(struct sockaddr_in);
+		}
+
+		ret = connect(client->transport.tcp.sock, client->broker,
+			      peer_addr_size);
+		if (ret < 0) {
+			(void)close(client->transport.tcp.sock);
+			return -errno;
+		}
 	}
+#if defined(CONFIG_SOCKS) && !defined(CONFIG_MQTT_LIB_TLS)
+	else {
+		const struct sockaddr *proxy = client->proxy;
 
-	ret = connect(client->transport.tcp.sock, client->broker,
-		      peer_addr_size);
-	if (ret < 0) {
-		(void)close(client->transport.tcp.sock);
-		return -errno;
+		client->transport.tcp.sock =
+			socks5_client_tcp_connect(proxy, broker);
```

Comment:
you would need to make dummy inline in the socks header file, if CONFIG_SOCKS isn't enabled